tests/dummy/app/templates/components/validated-input.hbs

```diff
@@ -5,10 +5,16 @@
   {{/if}}
 
   <div class="input-error">
-      {{#if showMessage}}
+      {{#if showErrorMessage}}
           <div class="error">
-              {{v-get model valuePath 'message'}}
+              {{get (v-get model valuePath 'error') 'message'}}
           </div>
       {{/if}}
+
+      {{#if showWarningMessage}}
+        <div class="warning">
+            {{get (v-get model valuePath 'warning') 'message'}}
+        </div>
+      {{/if}}
```

Comment:
My theory was:

1. Not to break current API since there could be other addons that are dependent that the messages array will return a collection of strings that are required.

2. I wanted to separate out warning and errors because errors are whats important. Warnings are optional and should affect the state of the model in any way. 

We could release 3.0 with API changes and think of a better way to handle this